CoreFeatures/Endpoint/EndpointExternalConfigurationFileReader/src/main/java/info/smart_tools/smartactors/endpoint/endpoint_external_configuration_file_reader/EndpointExternalConfigurationFileReader.java

```diff
@@ -0,0 +1,64 @@
+package info.smart_tools.smartactors.endpoint.endpoint_external_configuration_file_reader;
+
+import info.smart_tools.smartactors.base.exception.invalid_argument_exception.InvalidArgumentException;
+import info.smart_tools.smartactors.endpoint.iendpoint_external_configuration_reader.IEndpointExternalConfigurationReader;
+import info.smart_tools.smartactors.endpoint.iendpoint_external_configuration_reader.exception.EndpointExternalConfigurationReaderException;
+import info.smart_tools.smartactors.iobject.ifield_name.IFieldName;
+import info.smart_tools.smartactors.iobject.iobject.IObject;
+import info.smart_tools.smartactors.iobject.iobject.exception.ReadValueException;
+import info.smart_tools.smartactors.ioc.iioccontainer.exception.ResolutionException;
+import info.smart_tools.smartactors.ioc.ikey.IKey;
+import info.smart_tools.smartactors.ioc.ioc.IOC;
+import info.smart_tools.smartactors.ioc.key_tools.Keys;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Properties;
+
+/**
+ * {@link IEndpointExternalConfigurationReader} implementation for reading external configuration from file
+ */
+public class EndpointExternalConfigurationFileReader implements IEndpointExternalConfigurationReader {
+    private final Properties props;
+
+    /**
+     * This constructor loads external configuration from file that lays in configuration.pathToExternalConfiguration
+     * @param configuration                                 default endpoint configuration
+     * @throws EndpointExternalConfigurationReaderException if something went wrong while loading
+     */
+    public EndpointExternalConfigurationFileReader(final IObject configuration) throws EndpointExternalConfigurationReaderException {
+        try {
+            IKey fieldNameKey = Keys.getKeyByName(IFieldName.class.getCanonicalName());
+            IFieldName pathToExternalConfigurationFN = IOC.resolve(fieldNameKey, "pathToExternalConfiguration");
+
+            FileReader propsReader = new FileReader((String) configuration.getValue(pathToExternalConfigurationFN));
```

Comment:
I see FileReader creation, but i don't see call to method close()

Reply:
Fixed. I moved creating FileReader to the resources of try-catch block